elasticdl/python/master/evaluation_service.py

```diff
@@ -92,6 +95,28 @@ def get_evaluation_summary(self):
             ].items()
         }
 
+    def reset_metric_states(self):
+        """Resets all of the metric state variables."""
+        for metrics in self._metrics_dict.values():
+            for metric_inst in metrics.values():
+                metric_inst.reset_states()
+
+    @staticmethod
+    def _update_metric_by_small_chunk(
+        metric, labels, outputs, chunk_length=500
+    ):
+        """The metric updates state in a thread launched by grpc. The memory will
```

Comment:
Will the memory increase too if we split it into small chunks?

Reply:
Increase slowly. It is proportional to the size of the labels.